source/DSCResources/DSC_CMDistributionPointGroup/DSC_CMDistributionPointGroup.psm1

```diff
@@ -0,0 +1,311 @@
+$script:dscResourceCommonPath = Join-Path -Path $PSScriptRoot -ChildPath '..\..\Modules\DscResource.Common'
+$script:configMgrResourcehelper = Join-Path -Path $PSScriptRoot -ChildPath '..\..\Modules\ConfigMgrCBDsc.ResourceHelper'
+
+Import-Module -Name $script:dscResourceCommonPath
+Import-Module -Name $script:configMgrResourcehelper
+
+$script:localizedData = Get-LocalizedData -DefaultUICulture 'en-US'
+
+<#
+    .SYNOPSIS
+        This will return a hashtable of results.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER DistributionPointGroup
+        Specifies the Distribution Point Group name.
+#>
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $DistributionPointGroup
+    )
+
+    Write-Verbose -Message $script:localizedData.RetrieveSettingValue
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    $groupStatus = Get-CMDistributionPointGroup -Name $DistributionPointGroup
+
+    if ($groupStatus)
+    {
+        $dplist = Get-CMDistributionPoint -DistributionPointGroupName $DistributionPointGroup
+        $dpMembers = @()
+
+        foreach ($dp in $dplist)
+        {
+            $dpMembers += $dp.NetworkOSPath.SubString(2)
+        }
+
+        $group = 'Present'
+    }
+    else
+    {
+        $group = 'Absent'
+    }
+
+    return @{
+        SiteCode               = $SiteCode
+        DistributionPointGroup = $DistributionPointGroup
+        DistributionPoints     = $dpMembers
+        Ensure                 = $group
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER DistributionPointGroup
+        Specifies the Distribution Point Group name.
+
+    .PARAMETER DistributionPoints
+        Specifies an array of distribution points to match to the distribution point group.
+
+    .PARAMETER DistributionPointsToInclude
+        Specifies an array of distribution points to add to the distribution point group.
+
+    .PARAMETER DistributionPointsToExclude
+        Specifies an array of distribution points to remove from the distribution point group.
+
+    .PARAMETER Ensure
+        Specifies if the Distribution Point Group is to be present or absent.
+#>
+function Set-TargetResource
```

Comment:
Needs handling for if a Distro Point is specified for both include and exclude for a DP Group.